### DISCUSSION TITLE
关于luamake的clean的一个小问题

### DISCUSSION THREAD

**Discussion by funcman** [ejoy/ant#147] [General]
ant和luamake是过年期间clone的，当时的环境是VS2019，后来换了环境为VS2022。

luamake在这种情况下，会报找不到ninja，应该是因为通过env.lua的PATH找不到已被卸载的VS2019内置的ninja。

但是，用luamake clean，并不会清除掉已生成的env.lua等文件，依然会直接报找不到ninja。

此时需要手工删除build文件夹，才能使luamake恢复正常工作。

那么有没有什么恰当的方式应对这种特殊情况？或者手工删除build文件夹就是最恰当的方式？

问这个问题，主要还是想了解一下开发组对构建工具的一些想法和思维。

**Comment by actboy168** [comment]
正常情况下，luamake clean应该是会删掉env.lua的

**Reply by funcman** [reply]
确定了一下，luamake实际没这个问题，我没仔细看代码，搞错了，抱歉。

我一时忘记应该用luamake -mode release clean来清理release的相关文件了。
